### ISSUE TITLE
Problem with .setValue(LocalTime.now());  - Invalid regular expression

### ISSUE THREAD

**Issue by masbaehr** [vaadin/vaadin-time-picker-flow#15]
There seems to be a problem when using 

```java
TimePicker tp = new TimePicker();
tp.setValue(LocalTime.now());
```

![image](https://user-images.githubusercontent.com/38694044/51825253-a7865f80-22e4-11e9-9f48-16c9246101c3.png)

**Comment by svipal** [comment]
Hey, I'm working on this and I can't reproduce it.

**Comment by masbaehr** [comment]
Did you check the JS Console (Chrome) and are you using Vaadin 13.0.1?

Additional information:

![image](https://user-images.githubusercontent.com/38694044/54995058-2a1a5b00-4fc6-11e9-95cd-e764c1e2169d.png)

**Comment by svipal** [comment]
I cloned the repo and launched the demo with this card :
```
private void bugDemo2() {
        Div message = createMessageDiv("time-picker-bug-demo2");
        // begin-source-example
        // source-example-heading: Bug
        TimePicker timePicker = new TimePicker();
        timePicker.setValue(LocalTime.now());
        // end-source-example
        addCard("BugDemo2", timePicker, message);
    }
```
I'm using Firefox; there's nothing wrong in the console. Will retry with Chrome.

**Comment by svipal** [comment]
![image](https://user-images.githubusercontent.com/48034417/54995748-cdb83b00-4fc7-11e9-98d8-d2d2eac6ceb5.png)
Works fine for me in both bowsers with the current version.

**Comment by masbaehr** [comment]
Can you try to make a breakpoint at the spot i showed, and post the content of the text variable?
I have the suspicion that this is something related to the system locale. As i'm not in the US!

![image](https://user-images.githubusercontent.com/38694044/54995824-fe987000-4fc7-11e9-91ab-20923d6002b6.png)

**Comment by svipal** [comment]
I don't really know how to do that from the Vaadin time picker flow side.
Adding console.log calls in vaadin-time-picker.html in the vaadin-time-picker-1.2.0.jar used by the demo does nothing sadly. 

I'm in France btw.

**Comment by masbaehr** [comment]
What result do you get if you do the following on your system:

System.out.println(LocalTime.now());

I'm getting 13:35:02.325198200

**Comment by svipal** [comment]
I get 13:40:14.936
Are you using the latest version of vaadin-time-picker-flow ?
Well, I thought it might have been that from your example earlier.
Maybe your milliseconds are too many and that's why the \d{1,3} regex fails
If that is the case, we can just truncate the milliseconds in the regex instead.

**Comment by masbaehr** [comment]
Hm - it would be interesting to know why i'm getting a different precision then. Whats your JVM? I'm using Oracle Java JVM 11.0.2 on Win X64

I'm using vaadin 13.0.1 which bundles time-picker-flow 1.1.0

**Comment by svipal** [comment]
Good point.
Openjdk8 on Solus 3.9999 

can you update your vaadin-time-picker-html with this and test  ?

```
parseTime: text => {
                    // Parsing with RegExp to ensure correct format
                    const MATCH_HOURS = '(\\d|[0-1]\\d|2[0-3])';
                    const MATCH_MINUTES = '(\\d|[0-5]\\d)';
                    const MATCH_SECONDS = MATCH_MINUTES;
                    const MATCH_MILLISECONDS = '(\\d+)';//changed the regex to allow for any number of digits
                    const re = new RegExp(`^${MATCH_HOURS}(?::${MATCH_MINUTES}(?::${MATCH_SECONDS}(?:\\.${MATCH_MILLISECONDS})?)?)?$`);
                    const parts = re.exec(text);
                    if (parts) {
                      // Allows setting the milliseconds with hundreds and tens precision
                      if (parts[4]) {
                        //truncate the string.
                        parts[4] = parts[4].substring(0,3);
                        while (parts[4].length < 3) {
                          parts[4] += '0';
                        }
                      }
                      return {hours: parts[1], minutes: parts[2], seconds: parts[3], milliseconds: parts[4]};
                    }
                  },
```

**Comment by svipal** [comment]
just so we're clear on it, could you also try using the git master version of vaadin-time-picker-flow ?

**Comment by masbaehr** [comment]
https://stackoverflow.com/questions/52029920/localdatetime-now-has-different-levels-of-precision-on-windows-and-mac-machine

This is probably the cause then. so we should make sure the precision is truncated correctly before applying the regex.

the master version should be the latest which is used in v 13.0.1.

Maybe it would be even better to do this in the java-code of the component

**Comment by svipal** [comment]
Good job finding that. 
On it.

**Comment by svipal** [comment]
```
@Override
public void setValue(LocalTime value) {
    	LocalTime truncated_value = value.truncatedTo(ChronoUnit.MILLIS);
 	super.setValue(truncated_value);
}
```
Tested it and it works, I will make a PR. Send me your email address btw so we can share the bounty.

**Comment by masbaehr** [comment]
Huh what bounty are you talking about? ;-) you can contact me via mas.baehr@gmail.com

**Comment by pleku** [comment]
Closed with #28
